### PR TITLE
Some recovery ES backports

### PR DESCRIPTION
**Pull request by seut**
See commits.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java [hunk 1]
```diff
@@ -90,24 +101,29 @@
...
     private volatile TimeValue internalActionTimeout;
+    private volatile TimeValue internalActionRetryTimeout;
     private volatile TimeValue internalActionLongTimeout;
 
     private volatile ByteSizeValue chunkSize = DEFAULT_CHUNK_SIZE;
 
     public RecoverySettings(Settings settings, ClusterSettings clusterSettings) {
 
         this.retryDelayStateSync = INDICES_RECOVERY_RETRY_DELAY_STATE_SYNC_SETTING.get(settings);
+        this.maxConcurrentFileChunks = INDICES_RECOVERY_MAX_CONCURRENT_FILE_CHUNKS_SETTING.get(settings);
```
Comment (seut) [inline server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java:118]: Ah this is a conflict resolution mistake by me, the commit was already backported, see https://github.com/crate/crate/pull/11805/files/08c1cc149514157c7216ffb5d8c3344264feeb29..0316d453e6bffbccf5fe9dd9f851e0fa093e8554#diff-9e0f47bc6c623b00660868f85caccbbba17dd27c9b22113157b614330ed8641bR122. Will fix.

**Review by mkleen — APPROVED** [review]
Looks good, tough last backport. One minor thing we used to remove the number of the pr from the elastic, but I think it is ok the way it is. 

```
bp: Sending operations concurrently in peer recovery (#58018) 
```
```
bp: Sending operations concurrently in peer recovery 
```

**Comment by mkleen** [comment]
And thanks for the fixup.

File: server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java (hunk 1 above)
Comment (mkleen) [reply server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java:118]: 👍

**Pull request merged**